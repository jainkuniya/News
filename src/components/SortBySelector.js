//Import libraries
import React, { Component } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

const moreStyles = StyleSheet.create({
	containerStyle: {
		padding: 8,
		flexDirection: 'row',
		alignItems: 'center',
	},
	testStyle: {
		paddingLeft: 16,
		paddingRight: 16,
	},
	touchableStyle: {
		marginLeft: 16,
		marginRight: 16,
	},
	selectedStyle: {
		borderRadius: 4,
		borderWidth: 0.5,
		borderColor: '#000000',
	},
	nonSelectedStyle: {
		borderRadius: 4,
		borderWidth: 0.5,
		borderColor: '#d6d7da',
	},

});

export default class SortBySelector extends Component {
	render() {
		const { 
			containerStyle,
			testStyle, 
			selectedStyle, 
			nonSelectedStyle,
			touchableStyle,
		} = moreStyles;
		const { onSelect, currentSelected } = this.props;

		return (
			<View style={containerStyle}>
				<TouchableOpacity
					style={[
						touchableStyle, 
						(currentSelected === 'latest') ? selectedStyle : nonSelectedStyle
						]}
					activeOpacity={0.7}
					onPress={() => onSelect('latest')}
				>
					<Text style={testStyle}>Latest</Text>
				</TouchableOpacity>

				<TouchableOpacity
					style={[
						touchableStyle, 
						(currentSelected === 'top') ? selectedStyle : nonSelectedStyle
						]}
					activeOpacity={0.7}
					onPress={() => onSelect('top')}
				>
					<Text style={testStyle}>Top</Text>
				</TouchableOpacity>

				<TouchableOpacity
					style={[
						touchableStyle, 
						(currentSelected === 'popular') ? selectedStyle : nonSelectedStyle
						]}
					activeOpacity={0.7}
					onPress={() => onSelect('popular')}
				>
					<Text style={testStyle}>Popular</Text>
				</TouchableOpacity>
			</View>
		);
	}
}
