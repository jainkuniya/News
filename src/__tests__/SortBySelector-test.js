import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import SortBySelector from '../components/SortBySelector';

test('renders correctly', () => {
	const onSortSelect = (sort: string) => (console.log(sort));

	const tree = renderer.create(
		<SortBySelector
			onSelect={onSortSelect}
			currentSelected={'latest'}
		/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});
