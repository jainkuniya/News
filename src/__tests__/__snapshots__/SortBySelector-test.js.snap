// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 8,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#000000",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "marginLeft": 16,
        "marginRight": 16,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      Latest
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#d6d7da",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "marginLeft": 16,
        "marginRight": 16,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      Top
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#d6d7da",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "marginLeft": 16,
        "marginRight": 16,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      Popular
    </Text>
  </View>
</View>
`;
